sql/sql_expression_operations.cc

```diff
@@ -1171,8 +1171,12 @@ static CellValue binaryModulus(const CellValue & la, const CellValue & ra)
 struct BinaryPlusOp {
     static CellValue apply(const CellValue & l, const CellValue & r)
     {
-        if (l.empty() || r.empty())
+        if (l.empty()) {
             return l;
+        }
+        if (r.empty()) {
+            return r;
+        }
```

Comment:
In that case, the logic is more convoluted than necessary.  We can just do if (l.empty() or r.empty()) return CellValue();